File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
     let counter = 0;

    for (let i = 0; i < nums.length; i++){
        if(nums[i] === 0){
            counter ++
        } else {
            [nums[i - counter], nums[i]] = [nums[i], nums[i - counter]]
        }
    };
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0

        for r in range(1, len(nums)):
            if nums[l] == 0 and nums[r] != 0:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
            
            while l < r and nums[l] != 0:
                l += 1